src/Compilers/CSharp/Test/Syntax/Parsing/AwaitParsingTests.cs

```diff
@@ -1913,6 +1914,87 @@ public void BadStatementInSyncContext()
             EOF();
         }
 
+        [WorkItem(25410, "https://github.com/dotnet/roslyn/issues/25410")]
+        [Fact]
+        public void AwaitIdentifierExpressionStatementInSyncContextWithinAsyncMethod()
```

Comment:
There should probably be more tests to cover anonymous methods & local functions, but I didn't find any tests whatsoever with anonymous methods in this file. For local functions, there are some tests in a separate file. But I feel like this scenario should be tested with all 3 constructs and in one place.

Reply:
:bulb: It would be great if you could add some tests for the missing cases as well.

Reply:
done